tilavarauspalvelu/models/user/queryset.py

```diff
@@ -53,5 +54,18 @@ def should_deactivate_permissions(self) -> models.QuerySet:
             & models.Q(last_login__date__lt=cutoff)
         )
 
+    def anonymize_inactive_users(self) -> None:
+        cutoff = local_date() - datetime.timedelta(days=settings.ANONYMIZE_USER_IF_LAST_LOGIN_IS_OLDER_THAN_DAYS)
+
+        # All users who haven't logged in in a while but haven't been anonymized yet
+        users = self.exclude(
+            first_name=ANONYMIZED_FIRST_NAME,
+            last_name=ANONYMIZED_LAST_NAME,
+        ).filter(last_login__date__lt=cutoff)
```

Comment:
I realised that in production there are many users who don't have `last_login` set (because they precede the authentication change) but are less than 2 years, or even just about less than 1 year old... so need to also check that `user.date_joined < cutoff` lest they be prematurely anonymized.

Reply:
Corrected the filter.